pinot-tools/src/main/java/org/apache/pinot/tools/tuner/meta/manager/JsonFileMetaManagerImpl.java

```diff
@@ -0,0 +1,245 @@
+      _path = val;
+      return this;
+    }
+
+    /**
+     *
+     * @param val If this is set the already applied index will not be considered again
+     * @return this
+     */
+    @Nonnull
+    public Builder setUseExistingIndex(@Nonnull Boolean val) {
+      _useExistingIndex = val;
+      return this;
+    }
+
+    /**
+     *
+     * @param val For research and debug purpose only, the cardinality of /tableName/colName passed in will be considered as 1
+     * @return this
+     */
+    @Nonnull
+    public Builder setAdditionalMaskingCols(@Nonnull HashMap<String, HashSet<String>> val) {
+      _additionalMaskingCols = val;
+      return this;
+    }
+
+    @Nonnull
+    public JsonFileMetaManagerImpl build() throws FileNotFoundException {
+      return new JsonFileMetaManagerImpl(this).fetch();
+    }
+  }
+
+  /**
+   * fetch the metadata from file
+   * @return this
+   */
+  public JsonFileMetaManagerImpl fetch() throws FileNotFoundException {
+    File file = new File(this._path);
+    String metaBytes = "";
+    try {
+      metaBytes = FileUtils.readFileToString(file);
+    } catch (IOException e) {
+      LOGGER.error("Can't read json file! ", e);
+      throw new FileNotFoundException();
+    }
+    try {
+      JsonNode jnode = JsonUtils.stringToJsonNode(metaBytes);
+      _aggregatedMap = jnode.get(COL_META);
+      _segmentMap = jnode.get(SEGMENT_META);
+    } catch (IOException e) {
+      LOGGER.error("Can not parse Json file!");
+      throw new FileNotFoundException();
+    }
+    return this;
+  }
+
+  /**
+   * If there is already inverted index on a column
+   * @param tableNameWithoutType
+   * @param columnName
+   * @return
+   */
+  public boolean hasInvertedIndex(String tableNameWithoutType, String columnName) {
+    if (_additionalMaskingCols.getOrDefault(tableNameWithoutType, new HashSet<>()).contains(columnName)) {
+      return true;
+    }
+    if (_useExistingIndex) {
+      String _numHasInv = getColField(tableNameWithoutType, columnName, SUM_SEGMENTS_HAS_INVERTED_INDEX);
+      return _numHasInv != null && Integer.parseInt(_numHasInv) > 0;
+    }
+    return false;
+  }
+
+  public BigFraction getAverageNumEntriesPerDoc(String tableNameWithoutType, String columnName) {
+    try {
+      BigInteger entriesNumerator = new BigInteger(getColField(tableNameWithoutType, columnName, SUM_TOTAL_ENTRIES));
+      BigInteger entriesDenominator = new BigInteger(getColField(tableNameWithoutType, columnName, SUM_DOCS));
+      if (entriesNumerator.compareTo(entriesDenominator) < 0 || entriesDenominator.equals(BigInteger.ZERO)) {
+        throw new Exception();
+      }
+      return new BigFraction(entriesNumerator, entriesDenominator);
+    } catch (Exception e) {
+      LOGGER.trace("Set average entries to 1 {} {}", tableNameWithoutType, columnName);
+      return BigFraction.ONE;
+    }
+  }
+
+  public BigFraction getColumnSelectivity(String tableNameWithoutType, String columnName) {
```

Comment:
I think this is an important method - adding some comments/rationale around how sorted/cardinality signals are used would be helpful.